src/core/server/http/http_server.mocks.ts

```diff
@@ -140,10 +139,7 @@ type DeepPartialObject<T> = { [P in keyof T]+?: DeepPartial<T[P]> };
 function createRawRequestMock(customization: DeepPartial<Request> = {}) {
   const pathname = customization.url?.pathname || '/';
   const path = `${pathname}${customization.url?.search || ''}`;
-  const url = new URL(
-    formatUrl(Object.assign({ pathname, path, href: path }, customization.url)),
-    'http://localhost'
-  );
+  const url = new URL(path, 'http://localhost');
```

Comment:
I don't quite think this new implementation is equivalent. In the old version, any other properties of `customization.url` that weren't defined in the target object would be copied over, whereas in the new one we only get the path logic. I think we still need to make sure that we maintain the other properties of `customization.url`.